flash/vision/video/classification/data.py

```diff
@@ -0,0 +1,194 @@
+        test_transform: Optional[Union[Callable, Module, Dict[str, Callable]]] = None,
+        predict_transform: Optional[Union[Callable, Module, Dict[str, Callable]]] = None,
+    ):
+
+        super().__init__()
+        self.clip_sampler = clip_sampler
+        self.video_sampler = video_sampler
+        self.decode_audio = decode_audio
+        self.decoder = decoder
+
+    def load_data(self, data: Any, dataset: IterableDataset) -> EncodedVideoDataset:
+        ds: EncodedVideoDataset = labeled_encoded_video_dataset(
+            data,
+            self.clip_sampler,
+            video_sampler=self.video_sampler,
+            decode_audio=self.decode_audio,
+            decoder=self.decoder,
+        )
+        if self.training:
+            dataset.num_classes = len(np.unique([s[1]['label'] for s in ds._labeled_videos]))
+        return ds
+
+
+class VideoClassificationData(DataModule):
+    """Data module for Video classification tasks."""
+
+    preprocess_cls = VideoPreprocessPreprocess
+
+    @classmethod
+    def instantiate_preprocess(
+        cls,
+        clip_sampler: ClipSampler,
+        video_sampler: Type[Sampler],
+        decode_audio: bool,
+        decoder: str,
+        train_transform: Dict[str, Union[nn.Module, Callable]],
+        val_transform: Dict[str, Union[nn.Module, Callable]],
+        test_transform: Dict[str, Union[nn.Module, Callable]],
+        predict_transform: Dict[str, Union[nn.Module, Callable]],
+        preprocess_cls: Type[Preprocess] = None
+    ) -> Preprocess:
+        """
+        """
+        preprocess_cls = preprocess_cls or cls.preprocess_cls
+        preprocess: Preprocess = preprocess_cls(
+            clip_sampler, video_sampler, decode_audio, decoder, train_transform, val_transform, test_transform,
+            predict_transform
+        )
+        return preprocess
+
+    @classmethod
+    def from_folders(
+        cls,
+        train_folder: Optional[Union[str, pathlib.Path]] = None,
+        val_folder: Optional[Union[str, pathlib.Path]] = None,
+        test_folder: Optional[Union[str, pathlib.Path]] = None,
+        predict_folder: Union[str, pathlib.Path] = None,
+        clip_sampler: Union[str, ClipSampler] = "random",
+        clip_duration: float = 2,
+        video_sampler: Type[Sampler] = RandomSampler,
+        clip_sampler_kwargs: Dict[str, Any] = None,
+        decode_audio: bool = True,
+        decoder: str = "pyav",
+        train_transform: Optional[Union[str, Dict]] = 'default',
+        val_transform: Optional[Union[str, Dict]] = 'default',
+        test_transform: Optional[Union[str, Dict]] = 'default',
+        predict_transform: Optional[Union[str, Dict]] = 'default',
+        batch_size: int = 4,
+        num_workers: Optional[int] = None,
+        preprocess_cls: Optional[Type[Preprocess]] = None,
+        **kwargs,
+    ) -> 'DataModule':
+        """
+
+        Creates a VideoClassificationData object from folders of images arranged in this way: ::
```

Comment:
If we're using EncodedVideoDataset (which supports Kinetics, HMDB51 and UCF1010), this should be "...from folders of videos..."

Reply:
Yes.